app/client/views/settings/data/data.js

```diff
@@ -1,70 +1,138 @@
-  'click #export-data' (event, templateInstance) {
+  "click #export-data"(event, templateInstance) {
     // Set 'fetching data' variable to true
     templateInstance.fetchingData.set(true);
 
     // Export all data
-    Meteor.call('exportAllData', function (error, exportData) {
+    Meteor.call("exportAllData", function(error, exportData) {
       if (!error) {
         // Convert export data to string
-        const dataString = JSON.stringify(exportData)
+        const dataString = JSON.stringify(exportData);
 
         // Create binary object from data string
-        const exportBlob = new Blob([dataString], {type: 'application/json;charset=utf-8'});
+        const exportBlob = new Blob([dataString], {
+          type: "application/json;charset=utf-8"
+        });
 
         // Trigger save as prompt
-        FileSaver.saveAs(exportBlob, 'GeriLife-export.json');
+        FileSaver.saveAs(exportBlob, "GeriLife-export.json");
 
         // hide spinner
         templateInstance.fetchingData.set(false);
       }
     });
   },
 
-  "change .file-upload-input": function(event, templateInstance){
-    
+  "change .file-upload-input": function(event, templateInstance) {
     var file = event.currentTarget.files[0];
-    if(file.type!=="application/json"){
+    templateInstance.file.set(file);
+    if (file.type !== "application/json") {
+      showAlert();
       templateInstance.isFileInvalid.set(true);
-      return
-    } 
+
+      return;
+    }
+    templateInstance.isFileInvalid.set(false);
+  },
+
+  "click #importData": function(event, templateInstance) {
     var reader = new FileReader();
-    
+
     reader.onload = function(fileLoadEvent) {
-      console.log(reader.result)
-       Meteor.call('file-upload', reader.result);
+      Meteor.call("JSONFileImport", reader.result, function(data) {
+        Session.set("JSONFileImportResult", data);
+      });
     };
-    reader.readAsText(file);
- }
-
+    reader.readAsText(templateInstance.file.get());
+  },
+  "click #clear": function(event, templateInstance) {
+    _clear();
+  }
 });
 
 Template.dataSettings.helpers({
-  fetchingData () {
+  fetchingData() {
     // Get reference to template instance
     const templateInstance = Template.instance();
 
     // return current value of 'fetchingData'
     return templateInstance.fetchingData.get();
   },
 
-  isFileInvalid(){
+  isFileInvalid() {
     const templateInstance = Template.instance();
     return templateInstance.isFileInvalid.get();
+  },
+
+  shouldShowAlert() {
+    const templateInstance = Template.instance();
+    return templateInstance.shouldShowAlert.get();
+  },
+
+  file() {
+    const templateInstance = Template.instance();
+    return templateInstance.file.get();
+  },
+
+  isDisabled() {
+    const templateInstance = Template.instance();
+    return !(
+      templateInstance.isFileInvalid.get() || !templateInstance.file.get()
+    );
+  },
+
+  notnotA(a) {
+    return !!a;
+  },
+
+  processRes() {
```

Comment:
Just do this all in the method callback, and without the "not not". The code should be really clear and easy to understand.